### DISCUSSION TITLE
Include the max_line_size parameter also in the COPY command

### DISCUSSION THREAD

**Discussion by parvizfarnia** [duckdb/duckdb#14199] [Ideas]
Hello,

Based on the online documentation there is a parameter named `max_line_size` which is available for `read_csv` function:

https://duckdb.org/docs/data/csv/overview.html

This indeed can be very useful whenever there is a CSV file with very long rows (more than 2 500 000 characters in a single line). 

Please correct me if I'm wrong, but it seemed to me that apparently this parameter is not currently available in the `COPY` command. Unless there is a major technical constraint, could this parameter be included for both read and write operations via the `COPY` command?

(Currently in our environment we use DuckDB-1.0.0)

Thanks in advance

**Comment by Tishj** [comment]
Definitely seems to be working

```sql
statement error
COPY special_date(t, d) from 'data/csv/special_date.csv' (MAX_LINE_SIZE 5)
----
Maximum line size of 5 bytes exceeded
```

What is the problem you're describing?

**Reply by parvizfarnia** [reply]
Very strange! I just did a second test today and indeed it worked.

Last night, I was importing a huge CSV file having lines with more than 2 000 000 characters. At a moment, I got the following error message:

```
Maximum line size of 2097152 bytes exceeded. Actual Size:2297709 bytes.

Possible Solution: Change the maximum length size, e.g., max_line_size=2297710

  file=/home/training/queries.csv
  delimiter = | (Set By User)
  quote = " (Set By User)
  escape = " (Auto-Detected)
  new_line = \r\n (Auto-Detected)
  header = false (Set By User)
  skip_rows = 0 (Auto-Detected)
  comment = \0 (Auto-Detected)
  date_format =  (Auto-Detected)
  timestamp_format =  (Auto-Detected)
  null_padding=0
  sample_size=20480
  ignore_errors=false
  all_varchar=0
The Column types set by the user do not match the ones found by the sniffer.
Column at position: 0 Set type: VARCHAR Sniffed type: TIMESTAMP
Column at position: 5 Set type: VARCHAR Sniffed type: BIGINT
Column at position: 9 Set type: BIGINT Sniffed type: VARCHAR
```


Therefore I added a new line : `max_line_size 99999999` to my Python method whose task is to call the `COPY` command to import my CSV:

```
def read_csv(self):
    sql_stmt = "".join(
        [
            f"copy mytable from '{self.srcfile}' ",
            " with (",
            "format csv, ",
            "header FALSE, ",
            "delimiter '|', ",
            "quote '\"', ",
            "null '', ",
            "encoding UTF8, ",
            "max_line_size 99999999"
            ")",
        ]
    )
    self.db_conn.execute(sql_stmt)
```
    
But I got an error message indicating that the `max_line_size` was not available for the `COPY` command. Given that it works now, I must have made some mitake (spelling of the parameter may be?) when I wrote the code.

Therefore, I confirm as you indicated that this works and I apologize for my mistake. Thanks for your help and your time.
